chain/transaction.go

```diff
@@ -359,6 +362,92 @@ func (t *Transaction) Marshal(p *codec.Packer) error {
 	return t.marshal(p)
 }
 
+func (t *Transaction) MarshalJSON() ([]byte, error) {
+	type txJSON struct {
+		ID      ids.ID `json:"id"`
+		Actions []byte `json:"actions"`
+		Auth    []byte `json:"auth"`
+	}
+
+	actionsPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	err := t.Actions.marshalInto(actionsPacker)
+	if err != nil {
+		return nil, err
+	}
+	if err := actionsPacker.Err(); err != nil {
+		return nil, err
+	}
+	authPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	authPacker.PackByte(t.Auth.GetTypeID())
+	t.Auth.Marshal(authPacker)
+	if err := authPacker.Err(); err != nil {
+		return nil, err
+	}
+
+	hexActions, err := codec.Bytes(actionsPacker.Bytes()).MarshalText()
```

Comment:
We don't need to call `MarshalText()` on this. We can just pass in the `codec.Bytes` type directly and this will be handled in `json.Marshal`